tests/e2e/tests/product-gallery/inner-blocks/product-gallery-large-image/product-gallery-large-image.block_theme.side_effects.spec.ts

```diff
@@ -149,4 +149,62 @@ test.describe( `${ blockData.name }`, () => {
 			await expect( styleOnHover.transform ).toBe( '' );
 		} );
 	} );
+
+	test( 'Renders correct image when selecting a product variation in the Add to Cart With Options block', async ( {
+		page,
+		editorUtils,
+		pageObject,
+	} ) => {
+		await pageObject.addProductGalleryBlock( { cleanContent: false } );
+		await pageObject.addAddToCartWithOptionsBlock();
+
+		const block = await pageObject.getMainImageBlock( {
+			page: 'editor',
+		} );
+
+		await expect( block ).toBeVisible();
+
+		await editorUtils.saveTemplate();
+
+		await page.goto( blockData.productPage, {
+			waitUntil: 'commit',
+		} );
+
+		const largeImageBlockOnFrontend = await pageObject.getMainImageBlock( {
+			page: 'frontend',
+		} );
+
+		const largeImageElement = largeImageBlockOnFrontend.locator(
+			'.wc-block-woocommerce-product-gallery-large-image__image--active-image-slide'
+		);
+
+		const imageSourceForLargeImageElement =
+			await largeImageElement.getAttribute( 'src' );
+
+		const addToCartWithOptionsBlock =
+			await pageObject.getAddToCartWithOptionsBlock( {
+				page: 'frontend',
+			} );
+		const addToCartWithOptionsColorSelector =
+			addToCartWithOptionsBlock.getByLabel( 'Color' );
+		const addToCartWithOptionsSizeSelector =
+			addToCartWithOptionsBlock.getByLabel( 'Logo' );
+
+		await addToCartWithOptionsColorSelector.selectOption( 'Green' );
+		await addToCartWithOptionsSizeSelector.selectOption( 'No' );
+
+		const largeImageElementAfterSelectingVariation =
+			largeImageBlockOnFrontend.locator(
+				'.wc-block-woocommerce-product-gallery-large-image__image--active-image-slide'
+			);
+
+		const imageSourceForLargeImageElementAfterSelectingVariation =
+			await largeImageElementAfterSelectingVariation.getAttribute(
+				'src'
+			);
+
+		expect( imageSourceForLargeImageElement ).not.toEqual(
```

Comment:
What do you think instead of testing that they are different, we explicitly test for if the "green hoodie" is the one that is actually switched to? So that way we know selecting `Green, No` variation actually produces the correct image?

Something like this?
```
		expect(
			imageSourceForLargeImageElementAfterSelectingVariation
		).toContain( 'hoodie-green-1.jpg' );
```

Reply:
Great suggestion! I just edited the file and added your suggestion to it 🙌